### PR TITLE
Data plane verification

### LINKED ISSUE #4166: data plane messages need to be signed
#### Problem
 data plane protocol messages (e.g. repair requests) aren't signed, opening the protocol to spam

 #### Proposed Solution
 sign and verify all data plane protocol messages

- [ ] sign and verify per-blob repair requests
- [ ] sign and verify "back chain" repair requests

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Blobs on the data plane are not signed/verified

#### Summary of Changes
1) Add signature to blob
2) Sign the blob in broadcast
2) Verify signature in window service

Fixes #
Fixes #4166

### CODE REVIEW

File: core/src/cluster_info.rs [hunk 1]
```diff
@@ -44,6 +44,7 @@ use solana_sdk::pubkey::Pubkey;
 use solana_sdk::signature::{Keypair, KeypairUtil, Signable, Signature};
 use solana_sdk::timing::{duration_as_ms, timestamp};
 use solana_sdk::transaction::Transaction;
+use std::borrow::Cow;
```
Comment (sagar-solana) [inline core/src/cluster_info.rs]: 🐄

File: core/src/window_service.rs [hunk 2]
```diff
@@ -86,7 +91,10 @@ pub fn should_retransmit_and_persist(
         Some(bank) => leader_schedule_cache.slot_leader_at(blob.slot(), Some(&bank)),
     };
 
-    if blob.id() == *my_pubkey {
+    if !blob.verify() {
```
Comment (sagar-solana) [inline core/src/window_service.rs]: _maybe_ a good idea to file an issue to possibly offload this to gpu  ?

**Comment by sagar-solana** [comment]
seems ok to me

**Comment by garious** [comment]
Seems odd to see the Cows in this otherwise-correctness work and no benchmark. How'd you end up deciding to use those? What happens if you don't?

File: core/src/window_service.rs (hunk 2 above)
Comment (carllin) [reply core/src/window_service.rs]: https://github.com/solana-labs/solana/issues/4668

**Pull request merged**